pubsub/gcppubsub/gcppubsub.go

```diff
@@ -210,8 +211,27 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 
 // OpenSubscriptionURL opens a pubsub.Subscription based on u.
 func (o *URLOpener) OpenSubscriptionURL(ctx context.Context, u *url.URL) (*pubsub.Subscription, error) {
-	for param := range u.Query() {
-		return nil, fmt.Errorf("open subscription %v: invalid query parameter %q", u, param)
+	// Set subscription options to use defaults
+	o.SubscriptionOptions = SubscriptionOptions{
+		MaxBatchSize: defaultRecvBatcherOpts.MaxBatchSize,
+	}
+
+	for param, value := range u.Query() {
+		switch param {
+		case "max-batch-size":
```

Comment:
Please update the comments near line 182 to document this URL parameter.

See https://github.com/google/go-cloud/blob/master/blob/gcsblob/gcsblob.go#L204 for an example. Make sure it's clear that it only affects Subscriber.